src/ophys_etl/modules/segmentation/seed/seeder.py

```diff
@@ -0,0 +1,317 @@
+
+        Parameters
+        ----------
+        pixels (set): Set of pixel coordinates to exclude.
+
+        Returns
+        -------
+        None
+
+        """
+        dilated_pixels = dilated_coordinates(pixels,
+                                             self._exclusion_buffer)
+        self._excluded_pixels.update(dilated_pixels)
+
+    def __next__(self):
+        """returns the next valid seed
+
+        Notes
+        -----
+        A seed can be invalid either because it was too low on the list of
+        seeds sorted by metric (excluded_by_fraction) or if it was within
+        a buffer of an ROI and is in _excluded_pixels (excluded_by_roi)
+
+        """
+        if len(self._candidate_seeds) == 0:
+            raise StopIteration
+
+        next_seed = self._candidate_seeds.pop(0)
+
+        if next_seed['coordinates'] in self._excluded_pixels:
+            next_seed['exclusion_reason'] = "excluded by ROI"
+            self._excluded_seeds.append(next_seed)
+            return self.__next__()
+
+        self._provided_seeds.append(next_seed)
+        return next_seed['coordinates']
+
+    def select_seeds(self):
+        """method for selecting seeds. This method should be implemented
+        by subclasses.
+
+        Notes
+        -----
+        implementations of this method should populate `self._seeds`
+        as List[Seed]
+        """
+        raise NotImplementedError(f"class {type(self).__name__} does not "
+                                  "implement the method 'select_seeds()'")
+
+
+class ImageMetricSeeder(SeederBase):
+    """Finds seeds based on an image input and non-overlapping blocks
+    of that image
+
+    Parameters
+    ----------
+    keep_fraction: float
+        when sorted by the image pixel values, the top keep_fraction
+        seeds will be valid and the bottom (1.0 - keep_fraction) will
+        be marked as 'excluded_by_fraction'
+    seeder_grid_size: int
+        this follow the HNCcorr method of considering candidate seeds
+        from every non-overlapping seeder_grid_size x seeder_grid_size
+        subimage
+
+    Attributes
+    ----------
+    _keep_fraction
+        see Parameter 'keep_fraction'
+    _seeder_grid_size
+        see Parameter 'seeder_grid_size'
+    _seed_image
+        the input image to 'select_seeds' which is stored for access
+        by 'log_to_h5_group()'
+
+    Notes
+    -----
+    This class is largely inspired by HNCcorr and is written to be
+    usable with little modification by that algorithm.
+    https://github.com/hochbaumGroup/HNCcorr
+    Inheriting from 'Iterator' directly, adding attributes, a logging method,
+    and operating on any input image (separating out the computation of that
+    image) are the notable changes from that inspiration.
+
+    """
+    def __init__(self,
+                 keep_fraction: float = 0.4,
+                 seeder_grid_size: Optional[int] = None,
+                 *args, **kwargs):
```

Comment:
`*args` and `**kwargs` should probably be replaced with explicit parameters?

Reply:
I was wondering about this.
Really, I think I only need `**kwargs` and, because they are passed to `super`, any undefined arg will raise an exception.
So, explicitly defining base class args in every subclass seemed to be a lot of boilerplate.

Reply:
compromised by removing `*args` from passing through to base classes.